geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/ClusteringUtils.java

```diff
@@ -0,0 +1,409 @@
+			final String accumuloUser,
+			final String accumuloPassword,
+			final String namespace,
+			final String[] dimensionNames )
+			throws Exception {
+
+		final FeatureDataAdapter adapter = AnalyticFeature.createGeometryFeatureAdapter(
+				sampleDataTypeId,
+				dimensionNames,
+				4326);
+
+		final AccumuloOperations operations = basicAccumuloOperationsBuilder.build(
+				zookeeper,
+				accumuloInstance,
+				accumuloUser,
+				accumuloPassword,
+				namespace);
+
+		final AccumuloAdapterStore adapterStore = new AccumuloAdapterStore(
+				operations);
+
+		final ByteArrayId dbId = new ByteArrayId(
+				sampleDataTypeId);
+		if (!adapterStore.adapterExists(dbId)) {
+			adapterStore.addAdapter(adapter);
+			return adapter;
+		}
+		else {
+			return adapterStore.getAdapter(dbId);
+		}
+
+	}
+
+	public static DataAdapter[] getAdapters(
+			final String zookeeper,
+			final String accumuloInstance,
+			final String accumuloUser,
+			final String accumuloPassword,
+			final String namespace ) {
+		BasicAccumuloOperations ops;
+		try {
+			ops = new BasicAccumuloOperations(
+					zookeeper,
+					accumuloInstance,
+					accumuloUser,
+					accumuloPassword,
+					namespace);
+
+			final AccumuloAdapterStore adapterStore = new AccumuloAdapterStore(
+					ops);
+			final mil.nga.giat.geowave.store.CloseableIterator<DataAdapter<?>> it = adapterStore.getAdapters();
+			final List<DataAdapter> adapters = new LinkedList<DataAdapter>();
+			while (it.hasNext()) {
+				adapters.add(it.next());
+			}
+
+			final DataAdapter[] result = new DataAdapter[adapters.size()];
+			adapters.toArray(result);
+			return result;
+		}
+		catch (final AccumuloException e) {
+			LOGGER.error(
+					"cannot connect to GeoWave ",
+					e);
+		}
+		catch (final AccumuloSecurityException e) {
+			LOGGER.error(
+					"cannot connect to GeoWave ",
+					e);
+		}
+		return null;
+	}
+
+	public static Index[] getIndices(
+			final String zookeeper,
+			final String accumuloInstance,
+			final String accumuloUser,
+			final String accumuloPassword,
+			final String namespace ) {
+		BasicAccumuloOperations ops;
+		try {
+			ops = basicAccumuloOperationsBuilder.build(
+					zookeeper,
+					accumuloInstance,
+					accumuloUser,
+					accumuloPassword,
+					namespace);
+
+			final AccumuloIndexStore indexStore = new AccumuloIndexStore(
+					ops);
+			final mil.nga.giat.geowave.store.CloseableIterator<Index> it = indexStore.getIndices();
+			final List<Index> indices = new LinkedList<Index>();
+			while (it.hasNext()) {
+				indices.add(it.next());
+			}
+
+			final Index[] result = new Index[indices.size()];
+			indices.toArray(result);
+			return result;
+		}
+		catch (final AccumuloException e) {
+			LOGGER.error(
+					"cannot connect to GeoWave ",
+					e);
+		}
+		catch (final AccumuloSecurityException e) {
+			LOGGER.error(
+					"cannot connect to GeoWave ",
+					e);
+		}
+		return new Index[] {
+			IndexType.SPATIAL_VECTOR.createDefaultIndex()
+		};
+	}
+
```

Comment:
it seems everything below this line is only used by tests...do we need it here, my inclination is it could be code bloat?  (ie. suggestion is to move it to test)